test/e2e/aws_test.go

```diff
@@ -19,112 +19,438 @@ limitations under the License.
-	}
-	region = strings.TrimSpace(val)
-	return nil
-}
-
-var _ = Describe("AWS", func() {
+var _ = Describe("functional tests", func() {
 	var (
-		kindCluster kind.Cluster
+		namespace               string
+		clusterName             string
+		awsClusterName          string
+		cpMachinePrefix         string
+		cpAWSMachinePrefix      string
+		cpBootstrapConfigPrefix string
+		testTmpDir              string
 	)
 	BeforeEach(func() {
-		fmt.Fprintf(GinkgoWriter, "running in AWS region: %s\n", region)
-		kindCluster = kind.Cluster{
-			Name: "capa-test-" + util.RandomString(6),
-		}
-		kindCluster.Setup()
-	}, kindTimeout)
-
-	AfterEach(func() {
-		kindCluster.Teardown()
+		var err error
+		testTmpDir, err = ioutil.TempDir(suiteTmpDir, "aws-test")
+		Expect(err).NotTo(HaveOccurred())
+
+		namespace = "test-" + util.RandomString(6)
+		createNamespace(namespace)
+
+		clusterName = "test-" + util.RandomString(6)
+		awsClusterName = "test-infra-" + util.RandomString(6)
+		cpMachinePrefix = "test-" + util.RandomString(6)
+		cpAWSMachinePrefix = "test-infra-" + util.RandomString(6)
+		cpBootstrapConfigPrefix = "test-boot-" + util.RandomString(6)
 	})
 
-	Describe("control plane node", func() {
-		It("should be running", func() {
-			sess := getSession()
-			accountID := getAccountID(sess)
+	Describe("workload cluster lifecycle", func() {
+		It("It should be creatable and deletable", func() {
+			By("Creating an AWSCluster")
+			makeAWSCluster(namespace, awsClusterName)
+
+			By("Creating a Cluster")
+			makeCluster(namespace, clusterName, awsClusterName)
+
+			By("Ensuring Cluster Infrastructure Reports as Ready")
+			waitForClusterInfrastructureReady(namespace, clusterName)
+
+			By("Creating the initial Control Plane Machine")
+			awsMachineName := cpAWSMachinePrefix + "-0"
+			bootstrapConfigName := cpBootstrapConfigPrefix + "-0"
+			machineName := cpMachinePrefix + "-0"
+			createInitialControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Deploying CNI to created Cluster")
+			deployCNI(testTmpDir, namespace, clusterName, *cniManifests)
+			waitForMachineNodeReady(namespace, machineName)
+
+			By("Creating the second Control Plane Machine")
+			awsMachineName = cpAWSMachinePrefix + "-1"
+			bootstrapConfigName = cpBootstrapConfigPrefix + "-1"
+			machineName = cpMachinePrefix + "-1"
+			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
 
-			createKeyPair(sess)
-			createIAMRoles(sess, accountID)
+			By("Creating the third Control Plane Machine")
+			awsMachineName = cpAWSMachinePrefix + "-2"
+			bootstrapConfigName = cpBootstrapConfigPrefix + "-2"
+			machineName = cpMachinePrefix + "-2"
+			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName)
+
+			// TODO: Deploy a MachineDeployment
+			// TODO: Scale MachineDeployment up
+			// TODO: Scale MachineDeployment down
```

Comment:
Let's open some issues for these?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1160